browser/ui/views/playlist/playlist_bubble.h

```diff
@@ -0,0 +1,43 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_VIEWS_PLAYLIST_PLAYLIST_BUBBLE_H_
+#define BRAVE_BROWSER_UI_VIEWS_PLAYLIST_PLAYLIST_BUBBLE_H_
+
+#include "base/memory/weak_ptr.h"
+#include "brave/browser/ui/views/playlist/playlist_bubbles_controller.h"
+#include "ui/base/metadata/metadata_header_macros.h"
+#include "ui/views/bubble/bubble_dialog_delegate_view.h"
+
+namespace views {
+class View;
+class Widget;
+}  // namespace views
+
+namespace playlist {
+class PlaylistTabHelper;
+
+class PlaylistBubble : public views::BubbleDialogDelegateView {
+  METADATA_HEADER(PlaylistBubble, views::BubbleDialogDelegateView)
+
+ protected:
+  PlaylistBubble(views::View* anchor_view,
+                 base::WeakPtr<PlaylistTabHelper> tab_helper);
+
+  ~PlaylistBubble() override = 0;
+
+  // BubbleDialogDelegate:
+  void OnWidgetDestroyed(views::Widget* widget) override;
+
+  base::WeakPtr<PlaylistTabHelper> tab_helper_;
```

Comment:
I think it'd be better to move weak_ptr after |next_bubble_|.

Reply:
That's intentional here. In constructors I usually use the order:
1. stuff that are being passed to base class constructors (e.g. `views::View*`)
2. stuff corresponding to member variables that are being initialized in the constructor (`base::WeakPtr<PlaylistTabHelper>`)
3. rest, if any

And then there's the matching declaration order of members in the class layout — since `next_bubble_` is not being initialized with the help of a constructor parameter, I put it as the last member. Additionally, `next_bubble_` is always used at the _end_ of a bubble's lifetime.

Reply:
My bad, I thought it's weak_ptr_factory. 🤭